src/apps/product/domain/category/category.service.ts

```diff
@@ -0,0 +1,55 @@
+import { BaseService } from "../../../../librairies/services/base.service";
+import { CategoryRepository } from "../../data-access/category/category.repository";
+import { ICategory } from "../../data-access/category/category.interface";
+import BadRequestError from "../../../../config/error/bad.request.config";
+
+export class CategoryService extends BaseService {
+  private repository: CategoryRepository;
+
+  constructor() {
+    super("Category");
+    this.repository = new CategoryRepository();
+  }
+
+  async createCategory(data: Partial<ICategory>): Promise<ICategory> {
+    if (data.parentId) {
```

Comment:
Hello @FAEZ10 ici au lieu de `Partial<ICategory>` il serait mieux d'utiliser `CategoryCreateDTO` pour une validation strict et cohérente. Pour cela il faudrait passer req.body à `this.dataToDtoInstance` Il y a un exemple dans `PermissionController` merci